### PR TITLE
Tx/Util: ensure only integers are allowed as input

### LINKED ISSUE #1497: Passing decimal values to FeeMarketEIP1559Transaction generates invalid output silently
In the snippet attached below, if you pass decimal numbers to the FeeMarketEIP1559Transaction constructor it doesn't throw any errors, but alters the values silently, this can be very dangerous as the final payload when ran in a browser environment converts `33974229950.550003` -> `35624562130324685` which is a much larger number than what was originally passed, increasing the gas price.

```
const ethereumJsTx = require('@ethereumjs/tx') //  "@ethereumjs/tx": "^3.3.0",

const decimal_gas_fee = 33974229950.550003
const base_fee = 5 * 10 ** 9

const rawTx = {
  maxPriorityFeePerGas: decimal_gas_fee,
  maxFeePerGas: decimal_gas_fee + base_fee,
  chainId: 1,
  nonce: 0,
  gasPrice: null,
  gasLimit: 144000,
  to: '0x0000000000000000000000000000000000000000',
  value: 0,
  data: []
}

const tx = ethereumJsTx.FeeMarketEIP1559Transaction.fromTxData(rawTx)

// Browser environment
// maxFeePerGas: 45061746130324685, maxPriorityFeePerGas: 35624562130324685

// Node.js environment
// maxFeePerGas: 2123389402, maxPriorityFeePerGas: 2123389402

console.log(tx.maxFeePerGas.toString(), tx.maxPriorityFeePerGas.toString())
```

### PR DESCRIPTION
**Pull request by jochem-brouwer**
This PR closes #1497 

`intToHex` now explicitly checks that the input value can be used. This also means that consumers who use JS (not TypeScript) are ensured to get correct values.

 `@ethereumjs/util` also does not export the `intToHex` and `intToBuffer` methods from `ethjs-util` anymore.

### CODE REVIEW

File: packages/tx/test/eip1559.spec.ts
Comment (ryanio) [inline packages/tx/test/eip1559.spec.ts:32]: can we also try string decimal? `'10.1'`

File: packages/util/src/bytes.ts
Comment (ryanio) [inline packages/util/src/bytes.ts:15]: should we just do

```
return `0x${i.toString(16)}`
```

(no need to initialize a const?)

or also would this satisfy eslint:

```suggestion
  const hex = i.toString(16)
```

File: packages/util/src/bytes.ts
Comment (ryanio) [inline packages/util/src/bytes.ts:12]: should we add `|| !Number.isSafeInteger(i)`?

Comment (gabrocheleau) [inline packages/util/src/bytes.ts:12]: That last check `i % 1 !== 0` will evaluate to false when `i === 0`, which I don't think is what we want, since we want to allow 0.
What do you think of the following:
 ```suggestion
  if (typeof i !== 'number' || isNaN(i) || i === Infinity || i < 0 || (i !== 0 && i % 1 !== 0)) {
```

File: packages/tx/test/eip1559.spec.ts
Comment (jochem-brouwer) [reply packages/tx/test/eip1559.spec.ts:32]: I have not checked anything related with strings - good point because this will probably also produce garbage output.

File: packages/util/src/bytes.ts
Comment (jochem-brouwer) [reply packages/util/src/bytes.ts:12]: Yep good point!

Comment (gabrocheleau) [reply packages/util/src/bytes.ts:12]: Actually I take that back, I thought that `0 % 1` returned NaN for some reason.

Comment (jochem-brouwer) [reply packages/util/src/bytes.ts:12]: It seems like I can directly use this method instead of checking all these conditions :smile:

Comment (gabrocheleau) [reply packages/util/src/bytes.ts]: Cool, didn't know about that method!

I tested the conditions and they account for all of them except `i < 0`

Comment (jochem-brouwer) [reply packages/util/src/bytes.ts:12]: 0 % 1 = 0?

**Review by gabrocheleau — APPROVED** [review]

File: packages/util/src/bytes.ts
Comment (jochem-brouwer) [reply packages/util/src/bytes.ts]: Yup, just pushed the changes that accounts for it :)

File: packages/tx/test/eip1559.spec.ts
Comment (ryanio) [reply packages/tx/test/eip1559.spec.ts:34]: awesome, trying to think of some other last cases we can add here: `-1`, `-1.1`, `'-1.1'`

File: packages/util/src/bytes.ts
Comment (ryanio) [inline packages/util/src/bytes.ts:25]: harmless nit
```suggestion
```

File: packages/tx/test/eip1559.spec.ts
Comment (jochem-brouwer) [reply packages/tx/test/eip1559.spec.ts:34]: I'll add all cases which I also used in util :)

**Comment by jochem-brouwer** [comment]
Note: Setting `chainId` to `false` or `NaN` for EIP1559/AccessList still creates a valid tx.

File: packages/util/src/bytes.ts
Comment (gabrocheleau) [inline packages/util/src/bytes.ts:11]: Not a big deal, but wouldn't it be simpler to just:
`export function intToHex(i: number) { ... ` ?

File: packages/util/src/bytes.ts
Comment (gabrocheleau) [inline packages/util/src/bytes.ts:23]: Same as above

File: packages/tx/test/typedTxsAndEIP2930.spec.ts
Comment (gabrocheleau) [inline packages/tx/test/typedTxsAndEIP2930.spec.ts:103]: Perhaps also add `null` and `undefined`

Comment (jochem-brouwer) [reply packages/tx/test/typedTxsAndEIP2930.spec.ts:103]: `undefined` won't work since that's the default field of an object if an attribute does not exist (tx will then choose a default value)

File: packages/util/src/bytes.ts
Comment (jochem-brouwer) [reply packages/util/src/bytes.ts:11]: Maybe, but this file exports all methods like this :smile:

File: packages/tx/test/typedTxsAndEIP2930.spec.ts
Comment (jochem-brouwer) [reply packages/tx/test/typedTxsAndEIP2930.spec.ts:103]: `null` also fails tests

Comment (gabrocheleau) [reply packages/tx/test/typedTxsAndEIP2930.spec.ts:103]: Ok thanks for clarifying 🙂

**Comment by jochem-brouwer** [comment]
Okay this is super weird, browser tests fail on `MAX_SAFE_INTEGER+1` ??

**Comment by jochem-brouwer** [comment]
Oh lol, never mind, I always immediately quit when browser build starts. Fixed upon rebuilding.

**Comment by gabrocheleau** [comment]
> Okay this is super weird, browser tests fail on `MAX_SAFE_INTEGER+1` ??

Wow weird. Just guessing here... but might it be that `MAX_SAFE_INTEGER+1` results in a rounding error that makes it equal to  `MAX_SAFE_INTEGER` (which passes the Number.isSafeInteger check) ?

**Comment by jochem-brouwer** [comment]
Ok I have addressed everything noted so far, this should be OK :)

File: packages/util/test/bytes.spec.ts
Comment (gabrocheleau) [inline packages/util/test/bytes.spec.ts:307]:
```suggestion
  st.ok(intToBuffer(0).equals(Buffer.from('00', 'hex')), 'correctly converts 0 to a buffer')
  st.ok(intToBuffer(1).equals(Buffer.from('01', 'hex')), 'correctly converts 1 to a buffer')
```

File: packages/util/test/bytes.spec.ts
Comment (gabrocheleau) [inline packages/util/test/bytes.spec.ts:327]:
```suggestion
  st.ok(intToHex(0) == '0x0', 'correctly converts 0 to a hex string')
  st.ok(intToHex(1) == '0x1', 'correctly converts 1 to a hex string')
```

**Review by gabrocheleau — CHANGES_REQUESTED** [review]
typos

**Review by gabrocheleau — APPROVED** [review]
Looks good to me.

**Comment by jochem-brouwer** [comment]
Let's wait for one more review before merging.

**Review by ryanio — APPROVED** [review]
lgtm!

**Review by holgerd77 — APPROVED** [review]
This is actually such an insanely great and clean work - especially regarding the test cases - I honestly don't know what to say.

Thanks a lot Jochem! 👍 🙂

I'll directly prepare a release on Util and Tx.

**Comment by holgerd77** [comment]
Ah, and I've tested this locally, also in direct comparison with the `master` branch version.

File: packages/util/test/bytes.spec.ts
Comment (jochem-brouwer) [inline packages/util/test/bytes.spec.ts:304]: nit: function

File: packages/util/test/bytes.spec.ts
Comment (jochem-brouwer) [inline packages/util/test/bytes.spec.ts:324]: nit: function

**Review by Comandante318 — COMMENTED** [review]
?

**Pull request merged**
